src/draw/lv_draw.c

```diff
@@ -430,6 +449,9 @@ static bool is_independent(lv_layer_t * layer, lv_draw_task_t * t_check)
     LV_PROFILER_BEGIN;
     lv_draw_task_t * t = layer->draw_task_head;
 
+    if(_draw_info.unit_cnt <= 1)
```

Comment:
Removing this line would fix the problem, but it would be a woarkaround. The question is why the tasks are put in queue out of order?

Reply:
It's a great idea!

They shouldn't be put into the queue out of order. They are added as LVGL traverses the widgets.

See this:
![image](https://github.com/user-attachments/assets/3449fc9d-1d79-48f0-a4a7-254710173c96)
It seems multiple CPU cores are trying to draw the 2 rectangles at the same time. There is a little bit of scrollbar then the white fill will be stronger. 

However I've checked that in all test cases there is only 1 draw unit. Which is mistake and earlier we had tests with more draw units, but somehow it was removed. I've tried and the test are failing with more draw unit. I'll investigate tomorrow.

Reply:
If there are 2 cpu cores but just one draw unit, then is should be just one draw thread - hence no race condition.
If there are 2 core and 2 sw draw units (if configured DRAW_SW_UNIT_COUNT 2), then it should not pass the if condition.

So I do not know why this behavior happens.

Reply:
I'm suspecting what's happening. It's related to the runtime created internal `lv_layer`s. When LVGL needs to create a layer (for a rotated widget, clip_corner, etc). In this case, each layer has its own list of draw tasks and somehow it's being messed up. 

I couldn't figure out the exact reason, therefore I suggest reverting this line and merge this PR for v9.2. We can fix it later in v9.3.

Reply:
Ok, I will do so.